lib/src/value/calculation.dart

```diff
@@ -120,6 +124,226 @@ class SassCalculation extends Value {
     return SassCalculation._("max", args);
   }
 
+  /// Creates a `hypot()` calculation with the given [arguments].
+  ///
+  /// Each argument must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation]. It must be passed at least one argument.
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  static Value hypot(Iterable<Object> arguments) {
+    var args = _simplifyArguments(arguments);
+    if (args.isEmpty) {
+      throw ArgumentError("hypot() must have at least one argument.");
+    }
+    _verifyCompatibleNumbers(args);
+
+    var subtotal = 0.0;
+    var index = 0;
+    var first = arguments.first;
+    if (first is! SassNumber) return SassCalculation._("hypot", args);
+    for (var number in arguments) {
+      if (number is! SassNumber || number.hasUnit('%')) {
```

Comment:
Hm I did not understand if you mean updating the code or the spec adding `hypot()` should require known units. Since `hypot()` is the same as `min`, `max`, and `clamp` in the spec I suppose it doesn't require known units so I'm removing `number.hasUnit('%')`